webapp/src/main/java/com/box/l10n/mojito/rest/textunit/CustomZonedDateTimeDeserializer.java

```diff
@@ -0,0 +1,41 @@
+package com.box.l10n.mojito.rest.textunit;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import java.io.IOException;
+import java.time.*;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Deserialize an object into {@link ZonedDateTime}. The object can either be a number of
+ * milliseconds from 1970-01-01T00:00:00Z or a string representing a date following the ISO format,
+ * such as '2023-11-28T22:15:30.000Z'.
+ *
+ * @author hylston.barbosa
+ */
+public class CustomZonedDateTimeDeserializer extends JsonDeserializer<ZonedDateTime> {
```

Comment:
Why do we need this class now? Where is it being used? Can you add unit tests?

Reply:
This class is needed to be able to deserialize long values as ZonedDateTime. Joda DateTime had a constructor that was able to handle long values, but ZonedDateTime doesn't.

Reply:
What was breaking without it? Is there a native Jackson / spring solution for this?